File: src/main/java/app/askresume/api/extract/validator/ExtractValidator.java
package app.askresume.api.extract.validator;

import app.askresume.global.error.ErrorCode;
import app.askresume.global.error.exception.BusinessException;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ExtractValidator {

    @Value("${file.licensed.content}")
    private List<String> licensedContentTypes;

    public void validateContentType(String contentType) {
        List<String> contentTypes = licensedContentTypes.stream()
                .filter(type -> !type.equalsIgnoreCase(contentType)).toList();

        if (contentTypes.size() != 0)
            throw new BusinessException(ErrorCode.NOT_PERMITTED_CONTENT_TYPE);

    }
}

File: src/main/kotlin/app/askresume/api/extract/validator/ExtractValidator.kt
package app.askresume.api.extract.validator

import app.askresume.global.error.ErrorCode
import app.askresume.global.error.exception.BusinessException
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class ExtractValidator {

    @Value("\${file.licensed.content}")
    private lateinit var licensedContentTypes: List<String>

    fun validateContentType(contentType: String?) {
        val contentTypes = licensedContentTypes.stream()
            .filter { type -> !type.equals(contentType, ignoreCase = true) }.toList()

        if (contentTypes.size != 0)
            throw BusinessException(ErrorCode.NOT_PERMITTED_CONTENT_TYPE)
    }
}

